packages/data/src/types.ts

```diff
@@ -123,3 +128,9 @@ type SelectorsOf< Config extends AnyConfig > = Config extends ReduxStoreConfig<
 	: never;
 
 export type combineReducers = typeof reduxCombineReducers;
+
+export type Controls = {
+	select: typeof controls.select;
+	resolveSelect: typeof controls.resolveSelect;
+	dispatch: typeof controls.dispatch;
+};
```

Comment:
I'm wondering what the value is of duplicating these types internally here. If all we want is the type of the `controls` property of the store why not use this instead?

```ts
export type Controls = typeof controls;
```

That being said it feels like `controls` itself should possibly be typed as well and then we just need to re-export it here.

```ts
export type { Controls } from './controls';
```

What do you think?

Reply:
Also, controls are no longer used in core – [they were replaced by thunks](https://make.wordpress.org/core/2021/10/29/thunks-in-gutenberg/). Not a blocker for this PR, as controls are still a part of the bundle and I'm sure plenty of plugins still depend on them, but I wanted to make sure you're aware of that.